src/main/java/saps/scheduler/core/Scheduler.java

```diff
@@ -198,10 +199,16 @@ protected void submitTasks(List<SapsImage> selectedTasks) {
             updateStateInCatalog(task, nextState, SapsImage.AVAILABLE, SapsImage.NON_EXISTENT_DATA,
                     SapsImage.NONE_ARREBOL_JOB_ID,
                     "updates task [" + task.getTaskId() + "] state for " + nextState.getValue());
-            String arrebolJobId = submitTaskToArrebol(task, nextState);
-            updateStateInCatalog(task, task.getState(), SapsImage.AVAILABLE, SapsImage.NON_EXISTENT_DATA, arrebolJobId,
-                    "updates task [" + task.getTaskId() + "] with Arrebol job ID [" + arrebolJobId + "]");
-            addTimestampTaskInCatalog(task, "updates task [" + task.getTaskId() + "] timestamp");
+            try {
+                String arrebolJobId = submitTaskToArrebol(task, nextState);
+                updateStateInCatalog(task, task.getState(), SapsImage.AVAILABLE, SapsImage.NON_EXISTENT_DATA, arrebolJobId,
+                        "updates task [" + task.getTaskId() + "] with Arrebol job ID [" + arrebolJobId + "]");
+                addTimestampTaskInCatalog(task, "updates task [" + task.getTaskId() + "] timestamp");
+            } catch (Exception e) {
+                updateStateInCatalog(task, ImageTaskState.FAILED, SapsImage.AVAILABLE, SapsImage.NON_EXISTENT_DATA, SapsImage.NONE_ARREBOL_JOB_ID,
+                        "Changed task [" + task.getTaskId() + "] state to FAILED");
+                addTimestampTaskInCatalog(task, "updates task [" + task.getTaskId() + "] timestamp");
+            }
```

Comment:
what exactly these changes do?

Reply:
Try to submit the Task for the Arrebol. If successful, change the task state and save the Job ID. If it fails, set the task state to failed.